pkg/webhook/webhook.go

```diff
@@ -490,3 +511,49 @@ func (wh *webhook) validateCVC(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionRe
 		"configured for operations other than UPDATE")
 	return response
 }
+
+// verifyExistenceOfSnapshots returns error when there are volumesnapshots or volumesnapshotDatas
+// related to volume and if any error occured else return nil
+func (wh *webhook) verifyExistenceOfSnapshots(pvc *corev1.PersistentVolumeClaim) error {
+	volumeSnapshotList, err := wh.snapClientSet.
+		VolumesnapshotV1().
+		VolumeSnapshots(pvc.Namespace).
+		List(metav1.ListOptions{
+			LabelSelector: snapshotMetadataPVName + "=" + pvc.Spec.VolumeName,
+		})
+	if err != nil {
+		return errors.Wrapf(err, "failed to list snapshots related to volume: %s", pvc.Spec.VolumeName)
+	}
+	if len(volumeSnapshotList.Items) > 0 {
+		return errors.Errorf("pvc %s has '%d' snapshot(s)", pvc.Name, len(volumeSnapshotList.Items))
+	}
+	volumeSnapshotDataList, err := wh.getVolumeSnapshotDataList(pvc)
+	if err != nil {
+		return err
+	}
+	if len(volumeSnapshotDataList.Items) > 0 {
+		return errors.Errorf("pvc %s has '%d' snapshotdata(s)", pvc.Name, len(volumeSnapshotDataList.Items))
+	}
+	return nil
+}
+
+// getVolumeSnapshotDataList return list of volumesnapshotdata list related
+// to provided persistent volume claim
+func (wh *webhook) getVolumeSnapshotDataList(
+	pvc *corev1.PersistentVolumeClaim) (*snapshot.VolumeSnapshotDataList, error) {
+	pvcSnapshotDataList := &snapshot.VolumeSnapshotDataList{}
+	snapshotDataList, err := wh.snapClientSet.
+		VolumesnapshotV1().
+		VolumeSnapshotDatas().
+		List(metav1.ListOptions{})
```

Comment:
Not remember exactly , does `VolumeSnapshotData` not have any PV based  label ?

Reply:
The PR description already has an example to make easy reviews

Reply:
ok i can see that you have added the full SnapshotData resources ( it was not there before when i reviwed)

Reply:
metadata added later.